### PR TITLE
fix(core): Treat unverified wallets as non existant

### PR DESCRIPTION
**Pull request by amanCypherock**
Changes added:
- Refactoring of card delete flow
- Added clearing of wallet data from memory after wallet fetch operation in card delete flow
- Treat wallet unverified as a case of wallet not found
- fix unverified error code not being returned to caller.

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6203 Update create wallet and delete wallet flows to resolve apdu len error](https://app.clickup.com/t/9002019994/PRF-6203)

File: src/card_flows/card_flow_delete_wallet.c [hunk 1]
```diff
@@ -138,61 +149,83 @@ static bool check_wallet_already_deleted_on_card(
...
+          (CARD_OPERATION_ABORT_OPERATION == error_code ||
+           CARD_OPERATION_VERIFICATION_FAILED == error_code)) {
+        // In case wallet is not found on card or wallet verification fails,
+        // consider it as a success case as wallet is already deleted, not
+        // created on card or wallet on card does not match with the required
+        // verification.
         clear_core_error_screen();
+        handle_wallet_deleted_from_card(&delete_cfg);
+        buzzer_start(BUZZER_DURATION);
+        continue;
```
Comment (ujjwal-cyph) [inline src/card_flows/card_flow_delete_wallet.c:210]: please add a note as to why we cannot afford to wait for card removal.

**Review by ujjwal-cyph — APPROVED** [review]
LGTM

File: src/card_flows/card_flow_delete_wallet.c (hunk 1 above)
Comment (amanCypherock) [reply src/card_flows/card_flow_delete_wallet.c]: updated, also added other changes.

**Pull request merged**